tendrl/commons/objects/cluster/atoms/delete_cluster_details/__init__.py

```diff
@@ -0,0 +1,112 @@
+        cluster_tendrl_context = NS.tendrl.objects.ClusterTendrlContext(
+            integration_id=integration_id
+        ).load()
+
+        etcd_keys_to_delete = []
+        etcd_keys_to_delete.append(
+            "/clusters/%s" % integration_id
+        )
+        etcd_keys_to_delete.append(
+            "/alerting/clusters/%s" % integration_id
+        )
+        etcd_keys_to_delete.append(
+            "/indexes/tags/detected_cluster/%s" %
+            cluster_tendrl_context.cluster_id
+        )
+        etcd_keys_to_delete.append(
+            "/indexes/tags/detected_cluster_id_to_integration_id/%s" %
+            cluster_tendrl_context.cluster_id
+        )
+        etcd_keys_to_delete.append(
+            "/indexes/tags/provisioner/%s" % integration_id
+        )
+        etcd_keys_to_delete.append(
+            "/indexes/tags/tendrl/integration/%s" %
+            integration_id
+        )
+        nodes = NS._int.client.read(
+            "/clusters/%s/nodes" % integration_id
+        )
+        node_ids = []
+        for node in nodes.leaves:
+            node_id = node.key.split("/")[-1]
+            node_ids.append(node_id)
+            etcd_keys_to_delete.append(
+                "/alerting/nodes/%s" % node_id
+            )
+
+        # Find the alerting/alerts entries to be deleted
+        try:
+            cluster_alert_ids = NS._int.client.read(
+                "/alerting/clusters"
+            )
+            for entry in cluster_alert_ids.leaves:
+                ca_id = entry.key.split("/")[-1]
+                etcd_keys_to_delete.append(
+                    "/alerting/alerts/%s" % ca_id
+                )
+        except etcd.EtcdKeyNotFound:
+            # No cluster alerts, continue
+            pass
+
+        try:
+            node_alert_ids = NS._int.client.read(
+                "/alerting/nodes"
+            )
+            for entry in node_alert_ids.leaves:
+                na_id = entry.key.split("/")[-1]
+                etcd_keys_to_delete.append(
+                    "/alerting/alerts/%s" % na_id
+                )
+        except etcd.EtcdKeyNotFound:
+            # No node alerts, continue
+            pass
+
+        # Remove the cluster details
+        for key in list(set(etcd_keys_to_delete)):
+            try:
+                NS._int.client.delete(key, recursive=True)
+            except etcd.EtcdKeyNotFound:
+                logger.log(
+                    "debug",
+                    NS.publisher_id,
+                    {
+                        "message": "The key: %s not found for deletion" %
+                        key
+                    },
+                    job_id=self.parameters['job_id'],
+                    flow_id=self.parameters['flow_id'],
+                )
+                continue
+
+        # Load the gluster servers list and remove
+        # the cluster nodes
+        try:
+            gl_srvr_list = NS._int.client.read(
+                "/indexes/tags/gluster/server"
```

Comment:
So actually I am not removing the tag actually here. I am just removing the nodes of the cluster being unmanaged from the list and write the tag back with pending node ids

Reply:
I think we still need to improve this. read("/indexes/tags/gluster/server") gives you every gluster node irrespective of cluster id

Reply:
If we remove the nodes of currently unmanaging cluster, it shouldnt matter for other clusters. How you see issue in this? Also what you suggest if not this way?

Reply:
I would suggest we let the tendrl-gluster-integration clean up its own entry in /indexes/tags/gluster/server 

So when you shutdown the tendrl-gluster-integration service during unmanage you need to add this clean up of tags.

The reason I dont want to load /indexes/tags/gluster/server is because it adds overhead of going through each and every gluster node across all clusters

Reply:
So you mean before stopping tendrl-gluster-integration, we should invoke a flow to clean the tags. So in gluster-integration as well we would need to load the list /indexes/tags/gluster/server and remove it node-id from the list and write it back.

So effectively each node loads this tag from etcd, removes its node-id from the list and writes it back. In existing case we are loading once in server and all the cluster nodes entries are removed from the list and then written back. Here is just one read and write whereas in other case (suggested one) it would `no os nodes * (1 read + 1 write)`